src/libraries/System.Private.CoreLib/src/System/Environment.cs

```diff
@@ -18,6 +18,23 @@ public static partial class Environment
         /// </summary>
         internal static bool IsSingleProcessor => ProcessorCount == 1;
 
+        private static volatile sbyte s_privilegedProcess;
+
+        /// <summary>
+        /// Gets whether the current process is authorized to perform security-relevant functions.
+        /// </summary>
+        public static bool IsPrivilegedProcess
+        {
+            get
+            {
+                if (s_privilegedProcess == 0)
+                {
+                    s_privilegedProcess = IsAdminProcess() ? (sbyte)1 : (sbyte)-1;
+                }
+                return s_privilegedProcess > 0;
```

Comment:
```suggestion
                sbyte privilegedProcess = s_privilegedProcess;
                if (privilegedProcess == 0)
                {
                    s_privilegedProcess = privilegedProcess  = IsAdminProcess() ? (sbyte)1 : (sbyte)-1;
                }
                return privilegedProcess > 0;
```

Reply:
Done.

Could you please explain benefits of the pattern? I'd expect compiler does the wosk well.

Reply:
s_,privilegedProcess is marked volatile. There were two volatile reads when only one was necessary, and it being volatile restricts the compiler's ability to remove the read.